edge-modules/load-gen/Settings.cs

```diff
@@ -29,20 +29,31 @@ public class Settings
                     configuration.GetValue<ulong>("messageSizeInBytes", 1024),
                     configuration.GetValue("transportType", TransportType.Amqp_Tcp_Only),
                     configuration.GetValue("outputName", "output1"),
-                    configuration.GetValue("startDelay", TimeSpan.FromSeconds(2)));
+                    configuration.GetValue("testStartDelay", TimeSpan.FromMinutes(5)),
+                    configuration.GetValue("testRunDuration", TimeSpan.MaxValue),
+                    configuration.GetValue("trackingId", string.Empty));
             });
 
         Settings(
             TimeSpan messageFrequency,
             ulong messageSizeInBytes,
             TransportType transportType,
             string outputName,
-            TimeSpan startDelay)
+            TimeSpan testStartDelay,
+            TimeSpan testRunDuration,
+            string trackingId)
         {
-            this.MessageFrequency = Preconditions.CheckNotNull(messageFrequency);
+            Preconditions.CheckRange(messageFrequency.Ticks, 0);
+            Preconditions.CheckRange(testStartDelay.Ticks, 0);
+            Preconditions.CheckRange(testRunDuration.Ticks, 0);
+
             this.MessageSizeInBytes = Preconditions.CheckRange<ulong>(messageSizeInBytes, 1);
             this.OutputName = Preconditions.CheckNonWhiteSpace(outputName, nameof(outputName));
-            this.StartDelay = startDelay;
+
+            this.MessageFrequency = messageFrequency;
+            this.TestRunDuration = testRunDuration;
+            this.TestStartDelay = testStartDelay;
+            this.TrackingId = trackingId ?? string.Empty;
             this.TransportType = transportType;
```

Comment:
Recommend adding Preconditions for TransportType and other fields

Reply:
TransportType is enum and all values are valid here; messageFrequency, testStartDelay, and testDuration are already validated above.  TrackingId is allowed to be null or whitespace to compatible with loadGen in stress/long haul scenario.

Reply:
nit: Setting tracking id default empty string in constructor seems awkward to me as we are defining defaults in multiple places. I feel a cleaner mental model is configuration.getValue for defaults and constructor for input validation.